### DISCUSSION TITLE
Amazon short urls

### DISCUSSION THREAD

**Discussion by rcarrillodev** [timbru31/amazon-tag-remover#1246] [Q&A] [answered]
Will this extension prevent tag injection when the link is shortened by amazon?

Like amzn.to/123213

**Comment by timbru31** [comment, marked as answer]
Yes it does! By using the `blocking` feature of the [webRequest.onBeforeRequest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onBeforeRequest) the URL is altered before a connection is being made.
If my memory serves right then the amzn.to is resolved first, which answers with a 302 (the long URL with the tag in it). This URL is then sanitized before your browser does the connection.
